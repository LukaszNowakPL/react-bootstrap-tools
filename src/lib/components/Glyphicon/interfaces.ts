export interface GlyphiconProps {
  icon: string;
  className?: string;
}

export interface ConfigInterface {
  defaultComponentClassName: string;
}
