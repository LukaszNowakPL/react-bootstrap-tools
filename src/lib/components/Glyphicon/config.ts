import { ConfigInterface } from "./interfaces";

const config: ConfigInterface = {
  defaultComponentClassName: "material-icons"
};

export default config;
