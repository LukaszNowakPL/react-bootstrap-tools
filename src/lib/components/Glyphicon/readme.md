# Glyphicon

Component for rendering [Material icon](https://material.io/tools/icons/?style=baseline).

![Glyphicon example](../../../../readme_img/GlyphiconExample.png)

## Example of usage

```
import React from 'react';
import { Glyphicon } from 'react-bootstrap-tools';

const SimplestExample = () => (
    <Glyphicon icon="add" />
);

export default SimplestExample;
```

## Available props

- string `icon` Material icon id. Click [here](https://material.io/tools/icons/?style=baseline) to check available icons library.

## Available optional props

- string `className` Additional class name to custom your styling of an icon.
  Styling example for `NewStyle` value:<pre>
  .NewStyle {
  background-color: yellow;
  color: red;
  }</pre>
