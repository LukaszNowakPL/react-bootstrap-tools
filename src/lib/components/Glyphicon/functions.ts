import config from "./config";

/**
 * Returns className used to create an icon
 * @param className
 */
export const getClassName = (className?: string) => {
  const defaultClassName = config.defaultComponentClassName;

  if (className) return defaultClassName + " " + className;

  return defaultClassName;
};
