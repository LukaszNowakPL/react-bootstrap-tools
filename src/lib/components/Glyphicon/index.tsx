import * as React from "react";
import { GlyphiconProps } from "./interfaces";
import { getClassName } from "./functions";

const Glyphicon = (props: GlyphiconProps) => {
  const { icon, className } = props;

  const componentClassName: string = getClassName(className);

  return <i className={componentClassName}>{icon}</i>;
};

export default Glyphicon;
