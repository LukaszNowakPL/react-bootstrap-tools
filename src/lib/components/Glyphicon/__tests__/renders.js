import React from "react";
import { shallow } from "enzyme";
import Glyphicon from "../index";

describe("<Glyphicon />", () => {
  const customComponentProps = {
    icon: "testIcon"
  };

  it("renders", () => {
    const component = shallow(<Glyphicon {...customComponentProps} />);
    expect(component).toMatchSnapshot();
  });

  it("renders with provided className", () => {
    const component = shallow(
      <Glyphicon {...customComponentProps} className="testClassName" />
    );
    expect(component).toMatchSnapshot();
  });
});
