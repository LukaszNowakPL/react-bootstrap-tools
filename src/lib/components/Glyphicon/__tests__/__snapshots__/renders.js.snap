// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Glyphicon /> renders 1`] = `
<i
  className="material-icons"
>
  testIcon
</i>
`;

exports[`<Glyphicon /> renders with provided className 1`] = `
<i
  className="material-icons testClassName"
>
  testIcon
</i>
`;
